293- Add Banner to recruit for ScPCA-nf

nozomione opened the pull request:
## Issue Number
#293 

## Purpose/Implementation Notes
Added the banner to top of the portal for the ScPCA-nf recruitment.

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist
- [x] Lint and unit tests pass locally with my changes

Tested on Chrome & FF. 

## Screenshots
N/A

## Note
Introduced a new component `Banner` for a banner to make it more descriptive. Please let me know any suggestions, thank you.

nozomione commented:
I've tested the UI against Chrome and Firefox. 

Any other browsers that should be included in the cross browser testing, please let me know. Thank you!

davidsmejia commented on client/src/components/Banner.js:
```diff
@@ -0,0 +1,70 @@
+  children
+}) => {
+  const [showing, setShowing] = useState(true)
+  const BannerBox = styled(Box)`
```
I think that a lot of this can be done with grommet using `Text`, `Paragraph`, and `Box` attributes. Please try and move what is possible outside of custom css. Sometimes it's easier to layout text by wrapping it in a `Box`

nozomione replied:
Inline styles are now updated as Grommet props(note: CSS rule for `children` props will remain as inline). Thank you!

davidsmejia reviewed (dismissed):
Could you test for the fixed navigation? I noticed the header looks misplaced when scrolling down on larger pages.

nozomione commented:
> Could you test for the fixed navigation? I noticed the header looks misplaced when scrolling down on larger pages.

Adjusted, thank you!

nozomione commented:
**(update)**
- Added the `BannerContext` and its hook to control the state of `Banner` and height of the parent container of fixed elements. 
- Adjusted `Banner` component accordingly.

Thank you!

davidsmejia commented on client/src/components/Banner.js:
```diff
@@ -0,0 +1,73 @@
+    ${({ theme }) => css`
+      background-color: ${theme.global.colors[bgColor] ||
+      theme.global.colors.gradient[bgColor]};
+      > div {
```
can you remove this style rule, this would be the responsibility of the person defining `children` to implement

davidsmejia commented on client/src/components/Banner.js:
```diff
@@ -0,0 +1,73 @@
+  fontColor = 'white',
+  htmlIcon,
+  iconName,
+  iconSize,
```
can you set a default for this

davidsmejia commented on client/src/components/Banner.js:
```diff
@@ -0,0 +1,73 @@
+    `}
+  `
+  useEffect(() => {
+    setShowing(true)
```
instead of hard coding `true` can you make this a prop, add something like `open = true`

davidsmejia commented on client/src/components/Banner.js:
```diff
@@ -0,0 +1,73 @@
+  const { showing, setShowing, onBannerClose } = useBanner()
+  const BannerBox = styled(Box)`
+    ${({ theme }) => css`
+      background-color: ${theme.global.colors[bgColor] ||
```
I'm not sure why we would need this if we have a default value for `bgColor`

nozomione commented:
Completed, thank you!

nozomione commented:
@davidsmejia  Reviewed your update and push some changes. Pls review and let me know if any questions(great update on Icon/Link components - now nicely optimized), thank you David!

davidsmejia commented on client/src/components/Banner.js:
```diff
@@ -0,0 +1,69 @@
+
+export const Banner = ({
+  label,
+  bgColor = 'brand',
```
make this background instead of bgColor

davidsmejia commented on client/src/components/Banner.js:
```diff
@@ -0,0 +1,69 @@
+  bgColor = 'brand',
+  ctaLabel,
+  ctaLink,
+  fontColor = 'white',
```
make this color

davidsmejia commented on client/src/contexts/BannerContext.js:
```diff
@@ -0,0 +1,32 @@
+  const [showing, setShowing] = useState(false)
+  const [bannerHeight, setBannerHeight] = useState(0)
+
+  const handleHideBanner = () => {
```
lets keep these as hide and show to indicate they are changing the state
